WordPress/Classes/ViewRelated/Blog/My Site/MySiteViewController+JetpackInstall.swift

```diff
@@ -2,34 +2,66 @@ import UIKit
     }
 
-    /// Presents the Jetpack Login (Install) controller from a notification
-    @objc func launchJetpackInstallFromNotification(_ notification: NSNotification) {
-        guard let blog = blog else {
+    /// Presents the prompt to install Jetpack from a notification
+    @objc func promptJetpackPluginInstallFromNotification(_ notification: NSNotification) {
+        guard let blog = blog,
+            let promptSettings = notification.userInfo?["jetpackInstallPromptSettings"] as? JetpackInstallPromptSettings else {
             return
         }
 
-        let controller = JetpackLoginViewController(blog: blog)
-        controller.promptType = .installPrompt
+        let installPromptViewController = JetpackInstallPromptViewController(blog: blog)
+        let installPromptNavigationController = UINavigationController(rootViewController: installPromptViewController)
+        installPromptNavigationController.modalPresentationStyle = .fullScreen
 
-        let navController = UINavigationController(rootViewController: controller)
-        navController.modalPresentationStyle = .fullScreen
+        installPromptViewController.actionHandler = { [weak self] action in
+            switch action {
+            case .noThanks:
+                installPromptNavigationController.dismiss(animated: true)
+            case .install:
+                self?.presentJetpackInstall(on: installPromptNavigationController, blog: blog, jetpackInstallPromptSettings: promptSettings)
+            }
+        }
 
-        navigationController?.present(navController, animated: true)
+        navigationController?.present(installPromptNavigationController, animated: true)
+    }
 
-        controller.completionBlock = { [weak self] in
-            defer {
-                navController.dismiss(animated: true)
-            }
+    /// Presents the Jetpack plugin install flow on top of the prompt
+    private func presentJetpackInstall(
+        on controller: UINavigationController,
+        blog: Blog,
+        jetpackInstallPromptSettings: JetpackInstallPromptSettings
+    ) {
+        let jetpackLoginViewController = JetpackLoginViewController(blog: blog)
+        jetpackLoginViewController.promptType = .installPrompt
+        let navigationController = UINavigationController(rootViewController: jetpackLoginViewController)
+        navigationController.modalPresentationStyle = .fullScreen
 
-            guard let self = self else {
-                return
-            }
+        controller.present(navigationController, animated: true)
+
+        jetpackLoginViewController.completionBlock = { [weak self] in
+            jetpackInstallPromptSettings.setPromptWasDismissed(true, for: blog)
+
+            /// If Jetpack install + WP.com login flow is completed, the blog will be synced and jetpack.isConnected will be true
+            /// Dismissing the prompt
+            if let jetpack = blog.jetpack, jetpack.isConnected {
+                self?.syncBlogs()
```

Comment:
Would this try to sync WP.com blogs even if Jetpack is not connected to a WP.com account? My understanding from reading the code today is that `isConnected` means Jetpack is connected to WP.com, but not connected to a WP.com account.

While I tested the test case where the user cancels the installation flow just before connecting their account, I haven't yet had time to test if this code is executed or if it's a different flow.

Reply:
Yes. We do this syncing to know if the blog actually became `isConnected` or not, after canceling the flow.  

> My understanding from reading the code today is that isConnected means Jetpack is connected to WP.com, but not connected to a WP.com account.

Yes, this is what it means.